waku/waku_discv5.nim

```diff
@@ -14,71 +14,73 @@ import
   eth/p2p/discoveryv5/protocol
-import
-  ./node/peer_manager/peer_manager,
-  ./waku_core,
-  ./waku_enr
+import ./node/peer_manager/peer_manager, ./waku_core, ./waku_enr
 
 export protocol, waku_enr
 
-
 declarePublicGauge waku_discv5_discovered, "number of nodes discovered"
 declarePublicGauge waku_discv5_errors, "number of waku discv5 errors", ["type"]
 
 logScope:
   topics = "waku discv5"
 
-
 ## Config
 
 type WakuDiscoveryV5Config* = object
-    discv5Config*: Option[DiscoveryConfig]
-    address*: IpAddress
-    port*: Port
-    privateKey*: eth_keys.PrivateKey
-    bootstrapRecords*: seq[waku_enr.Record]
-    autoupdateRecord*: bool
-
+  discv5Config*: Option[DiscoveryConfig]
+  address*: IpAddress
+  port*: Port
+  privateKey*: eth_keys.PrivateKey
+  bootstrapRecords*: seq[waku_enr.Record]
+  autoupdateRecord*: bool
 
 ## Protocol
 
-type WakuDiscv5Predicate* = proc(record: waku_enr.Record): bool {.closure, gcsafe, raises: [].}
+type WakuDiscv5Predicate* =
+  proc(record: waku_enr.Record): bool {.closure, gcsafe, raises: [].}
 
 type WakuDiscoveryV5* = ref object
-    conf: WakuDiscoveryV5Config
-    protocol*: protocol.Protocol
-    listening*: bool
-    predicate: Option[WakuDiscv5Predicate]
-    peerManager: Option[PeerManager]
-    topicSubscriptionQueue: AsyncEventQueue[SubscriptionEvent]
-
-proc shardingPredicate*(record: Record, bootnodes: seq[Record] = @[]): Option[WakuDiscv5Predicate] =
+  conf: WakuDiscoveryV5Config
+  protocol*: protocol.Protocol
+  listening*: bool
+  predicate: Option[WakuDiscv5Predicate]
+  peerManager: Option[PeerManager]
+  topicSubscriptionQueue: AsyncEventQueue[SubscriptionEvent]
+
+proc shardingPredicate*(
+    record: Record, bootnodes: seq[Record] = @[]
+): Option[WakuDiscv5Predicate] =
   ## Filter peers based on relay sharding information
-  let typedRecord = record.toTyped().valueOr:
-    debug "peer filtering failed", reason=error
-    return none(WakuDiscv5Predicate)
-
-  let nodeShard = typedRecord.relaySharding().valueOr:
-    debug "no relay sharding information, peer filtering disabled"
-    return none(WakuDiscv5Predicate)
-  
+  let typedRecord =
+    record.toTyped().valueOr:
+      debug "peer filtering failed", reason = error
+      return none(WakuDiscv5Predicate)
+
+  let nodeShard =
+    typedRecord.relaySharding().valueOr:
+      debug "no relay sharding information, peer filtering disabled"
+      return none(WakuDiscv5Predicate)
+
   debug "peer filtering updated"
 
   let predicate = proc(record: waku_enr.Record): bool =
-      bootnodes.contains(record) or # Temp. Bootnode exception
-      (record.getCapabilities().len > 0 and #RFC 31 requirement
-      nodeShard.shardIds.anyIt(record.containsShard(nodeShard.clusterId, it))) #RFC 64 guideline
+    bootnodes.contains(record) or # Temp. Bootnode exception
+    (
+      record.getCapabilities().len > 0 and
+      nodeShard.shardIds.anyIt(record.containsShard(nodeShard.clusterId, it))
+    ) #RFC 64 guideline
```

Comment:
Inline comments are going to be scrambled. Not sure what to do about that.

Reply:
Good catch!
That sounds more like a tiny bug. I think we cannot merge until this is sorted out. I will submit it